### ISSUE TITLE
TaxLevelCode seems to be a list, not singleton

### ISSUE THREAD

**Issue by blaggacao** [nubark/facturark#10]
If I'm not completely mistaken, I identified the `TaxLevelCode` as a list type, so:

```json
"party_tax_scheme": {
    "tax_level_code": "0"
}, 
```
should become
```json
"party_tax_scheme": {
    "tax_level_code": [ 
        "0" 
    ]
}, 
```

**Comment by tebanep** [comment]
In reality, we are not following the exact semantics of the xsd, mainly in favour of simplifying the json document construction in the context of the DIAN use of UBL. We could open the discussion about the suitability of this approach, as there could be corner cases where the exact semantics are needed.

**Comment by blaggacao** [comment]
If the minimum requirement is satisfied, there is indeed no need. However, as far as I can see, Carvajal is asking for a list of `TaxLevelCode`. We can close that for now, and I'll reopen if I find a violation of some minimum requirement. Full UBL2.0 support might be interesting in some more distant future :-)

**Comment by tebanep** [comment]
Closing this issue in the meanwhile, then. 👍

**Comment by blaggacao** [comment]
I provided a proposal, therefore reopening. Furthermore, it seems I depend on proper semantics according to my chosen data layout.

**Comment by blaggacao** [comment]
Ver tambien (p. 87, anexo 01) :
 Si la factura electrónica será utilizada como soporte de un crédito fiscal por el adquiriente de los
bienes o servicios, entonces deberán incluirse todas las informaciones del RUT, en forma similar a
las que se diligencian para el vendedor de los B / S.
 Si la factura electrónica será registrada en el REFEL, entonces deberán incluirse las informaciones
que hayan sido reglamentadas por el MinCIT

**Comment by blaggacao** [comment]
I understand that `PartyLegalEntity` is the space of camara de comercio and that `PartyTaxScheme` is the space of national, municipal and other tax identification data sets (like RUT, but not limited to it).
"deberán incluirse todas las informaciones del RUT" this means a lot more fields seem factually obligatory... However, not sure if this spec is really worth the name "spec" and maybe it's up to interpretation, still.

**Comment by blaggacao** [comment]
So a minimal `PartyTaxScheme` would look like:
```json
"party_tax_scheme": [{
    "registraion_name": "Name Name",
    "company_id": "978645312"
    "registraion_address": {"..." : "..."},
    "tax_level_code": ["2"]
},
{}
]
   // Note: first item is national tax register (=DIAN)
   // Note: Second and more items can be municipal, regional, etc.
```

**Comment by tebanep** [comment]
> I understand that `PartyLegalEntity` is the space of camara de comercio and that `PartyTaxScheme` is the space of national, municipal and other tax identification data sets (like RUT, but not limited to it).
> "deberán incluirse todas las informaciones del RUT" this means a lot more fields of the `PartyTaxScheme` seem _factually_ obligatory... However, not sure if this spec is really worth the name "spec" and maybe it's up to interpretation, still.
> 
> I think it would be wise (and future proof) to support the full scheme, in order not to run into any trouble later or being short on the data layout.

Ok. I'm convinced. So if I understand correctly, we need an array of tax schemes. I'll check it out an make a pull request.

**Comment by tebanep** [comment]
@blaggacao I have just checked the XSD and this is what I found:

```
 <complexType name="PartyTaxSchemeType">
        <sequence>
            <element ref="cbc:RegistrationName" minOccurs="0" maxOccurs="1">
                <annotation>
                    <documentation xml:lang="es">
						Opcional no usado por la DIAN, las partes pueden
						definir un significado o simplemente omitirlo
                    </documentation>
                </annotation>
            </element>
            <element ref="cbc:CompanyID" minOccurs="0" maxOccurs="1">
                <annotation>
                    <documentation>
						Opcional no usado por la DIAN, las partes pueden
						definir un significado o simplemente omitirlo
                    </documentation>
                </annotation>
            </element>
            <element ref="cbc:TaxLevelCode" minOccurs="1" maxOccurs="1">
                <annotation>
                    <documentation>
						2.2.3.8 - Régimen: Régimen al que pertenece.
						Debe referenciar a una lista de códigos con los
						valores correspondientes, por ejemplo: • Común •
						Simplificado • No aplica
                    </documentation>
                </annotation>
            </element>
            <element ref="cbc:ExemptionReasonCode" minOccurs="0" maxOccurs="1">
                <annotation>
                    <documentation xml:lang="es">
						Opcional no usado por la DIAN, las partes pueden
						definir un significado o simplemente omitirlo
                    </documentation>
                </annotation>
            </element>
            <element ref="cbc:ExemptionReason" minOccurs="0" maxOccurs="1">
                <annotation>
                    <documentation xml:lang="es">
						Opcional no usado por la DIAN, las partes pueden
						definir un significado o simplemente omitirlo
                    </documentation>
                </annotation>
            </element>
            <element ref="cac:RegistrationAddress" minOccurs="0" maxOccurs="1">
                <annotation>
                    <documentation xml:lang="es">
						Opcional no usado por la DIAN, las partes pueden
						definir un significado o simplemente omitirlo
                    </documentation>
                </annotation>
            </element>
            <element ref="cac:TaxScheme" minOccurs="1" maxOccurs="1">
                <annotation>
                    <documentation xml:lang="es">
						No usado por la DIAN, las partes pueden definir
						un significado o simplemente poner un elemento
						vacío ya que es obligatorio en UBL
                    </documentation>
                </annotation>
            </element>
        </sequence>
    </complexType>
```

The important part is this one:

<element ref="cbc:TaxLevelCode" minOccurs="1" **maxOccurs="1"**>

As you can see, the requirement is to provide **exactly one TaxLevelCode**.

Am I missing something?

**Comment by blaggacao** [comment]
What I've found (in `XSD/r1/DIAN_UBL.xsd` downloaded [here](https://www.dian.gov.co/fizcalizacioncontrol/herramienconsulta/FacturaElectronica/InformacionTecnica/Paginas/documentosapoyo.aspx): ):
```xml
<element ref="tns:PartyTaxScheme" minOccurs="1" maxOccurs="unbounded">
<annotation>
<documentation> Estructura necesaria para describir el régimen </documentation>
</annotation>
</element>
```

So effectively, we (can) have multiple PartyTaxScheme. But I also see what you copied. 

Then, again, reading [Anexo 001 V2](https://www.dian.gov.co/fizcalizacioncontrol/herramienconsulta/FacturaElectronica/Factura%20Electrnica/Anexo%20Tecnico%20001%20Formatos%20de%20los%20Documentos%20XML%20de%20Fac-e%20V2.pdf) For example at page 80 I read:
```xml
<!-- El fragmento
/fe:Invoice/fe:AccountingSupplierParty/fe:Party/fe:PartyTaxScheme[1..n]/cbc:TaxLevelCode se repetirá tantas veces ("n") como
la cantidad de valores registrados en el RUT. - Observe la conformación de otros fragmentos más adelante. --><!-- vea el archivo
Invoice__AdditionalAccountID.lst.html -->
```
Looking at `fe:PartyTaxScheme[1..n]` and referring `n` in the text I understand it should be formatted something like this (pseydo XML):

```xml
<fe:PartyTaxScheme>
  <cbc:TaxLevelCode>1</cbc:TaxLevelCode>
</fe:PartyTaxScheme>

<fe:PartyTaxScheme>
  <cbc:TaxLevelCode>2</cbc:TaxLevelCode>
</fe:PartyTaxScheme>

<fe:PartyTaxScheme>
  <cbc:TaxLevelCode>3</cbc:TaxLevelCode>
</fe:PartyTaxScheme>

<fe:PartyTaxScheme>
  <cbc:TaxLevelCode>4</cbc:TaxLevelCode>
</fe:PartyTaxScheme>
```

**Comment by blaggacao** [comment]
Ok, I validate my assumption with [this exemplification](file:///home/blaggacao/.cache/.fr-BUxTEX/f-s0001_mandatarioMandante654321_5A_cdSha256_11-firmado-SHA512.xml) (note the indented repetition): 
```xml
<fe:AccountingSupplierParty>
<cbc:AdditionalAccountID>1</cbc:AdditionalAccountID>
<fe:Party>
<cac:PartyIdentification>
<cbc:ID schemeAgencyID="195" schemeAgencyName="CO, DIAN (Direccion de Impuestos y Aduanas Nacionales)">700085464</cbc:ID>
</cac:PartyIdentification>
<cac:PartyName>
<cbc:Name>
nombre establecimiento del punto de venta || nombre mercantil del punto de venta
</cbc:Name>
</cac:PartyName>
<fe:PhysicalLocation>
<cbc:Description>ubicación del punto de venta</cbc:Description>
<fe:Address>
<cbc:Department>Bogota</cbc:Department>
<cbc:CitySubdivisionName>Centro</cbc:CitySubdivisionName>
<cbc:CityName>Bogota</cbc:CityName>
<cac:AddressLine>
<cbc:Line>carrera 7 Nº 6C - 82</cbc:Line>
</cac:AddressLine>
<cac:Country>
<cbc:IdentificationCode>CO</cbc:IdentificationCode>
</cac:Country>
</fe:Address>
</fe:PhysicalLocation>
  <fe:PartyTaxScheme>
  <cbc:RegistrationName>nombre del mandatario en el RUT</cbc:RegistrationName>
  <cbc:CompanyID schemeAgencyID="195" schemeAgencyName="CO, DIAN (Direccion de Impuestos y Aduanas Nacionales)" schemeDataURI="http://www.dian.gov.co RUT" schemeID="31" schemeName="NIT con DV del mandatario " schemeVersionID="'Tipos de documentos de identidad'">900373125</cbc:CompanyID>
  <cbc:TaxLevelCode listName="TIPOS OBLIGACIONES-RESPONSABILIDADES:2016" listSchemeURI="http://www.dian.gov.co Anexo-1" name="Ventas régimen común">O-11</cbc:TaxLevelCode>
  <cac:TaxScheme/>
  </fe:PartyTaxScheme>
  <fe:PartyTaxScheme>
  <cbc:TaxLevelCode listName="TIPOS PERSONA:2016" listSchemeURI="http://www.dian.gov.co Anexo-1" name="Persona Jurídica">1</cbc:TaxLevelCode>
  <cac:TaxScheme/>
  </fe:PartyTaxScheme>
  <fe:PartyTaxScheme>
  <cbc:TaxLevelCode listName="TIPOS REPRESENTACIÓN:2016" listSchemeURI="http://www.dian.gov.co Anexo-1" name="Mandatario">R-16-PJ</cbc:TaxLevelCode>
  <cac:TaxScheme/>
  </fe:PartyTaxScheme>
<fe:PartyLegalEntity>
<cbc:RegistrationName>
nombre del mandatario en la Cámara de Comercio o equivalente
</cbc:RegistrationName>
<cac:RegistrationAddress>
<cbc:CitySubdivisionName>barrio del mandatario</cbc:CitySubdivisionName>
<cbc:CityName>ciudad del mandatario</cbc:CityName>
<cbc:PostalZone>zona postal del mandatario</cbc:PostalZone>
<cbc:CountrySubentity>departamento del mandatario</cbc:CountrySubentity>
<cac:AddressLine>
<cbc:Line>dirección del mandatario</cbc:Line>
</cac:AddressLine>
<cac:Country/>
</cac:RegistrationAddress>
</fe:PartyLegalEntity>
</fe:Party>
</fe:AccountingSupplierParty>
```

**Comment by tebanep** [comment]
Perfect, I will make a pull request with the updated Party node

**Comment by tebanep** [comment]
I think this has been fixed already in the latest pull request (#16). What do you think @blaggacao?

**Comment by blaggacao** [comment]
That seems correct to me, thank you! Just in time to support the data layout, I have. :+1:
